docker/build.sh

```diff
@@ -1,54 +1,171 @@
 #!/usr/bin/env bash
-# Build a docker image for specific framework. Optionally `all` frameworks can be built.
-# Ex: `bash docker/build.sh pytorch`
-
-if [ "$#" == 1 ]; then
-    dev=""
-elif [ "$#" == 2 ]; then
-    if [[ "$2" != "--dev" && "$2" != "dev" ]]; then
-        echo "ERROR: argument 2 needs to be \`dev\` or \`--dev\`, not \`$2\`"
-        exit 1
-    fi
-    dev="-dev"
-else
-    echo "Usage: bash docker/build.sh <framework> [dev]"
-    echo "    <framework> be \`tf1\`, \`tf2\`, \`pytorch\`, \`pytorch-deepspeech\`, or \`all\`"
-    exit 1
+
+Help()
+{
+  # Display Help
+  echo "----------------------------------------------------------------------------------------"
+  echo "                   Armory Docker Build Script                       "
+  echo
+  echo "This build script helps to build the docker images necessary for armory exectuion "
+  echo "(in docker mode).  The primary purpose of this script is to build the images from "
+  echo "scratch and to be used by the armory CI toolchain.  If modifications are necessary"
+  echo "you can use \`--dry-run\` to get the \`docker build...\` commands directly and "
+  echo "modify them as necessary. "
+  echo ""
+  echo "Armory uses scm versioning from git, therefore if you want to re-build containers "
+  echo "from a stable armory release, simply checkout that tag and then run this script"
+  echo ""
+  echo "Syntax: build.sh [-f|nc|dr|v|h]"
+  echo ""
+  echo "REQUIRED"
+  echo "-f | --framework            Select Which framework to target:"
+  echo "                                [all|base|tf2|pytorch|pytorch-deepspeech]"
+  echo ""
+  echo "OPTIONAL"
+  echo "-t | --tag                  Specify Additional Tag to apply to each image"
+  echo "-nc | --no-cache            Build Images \`Clean\` (i.e. using --no-cache)"
+  echo "-dr | --dry-run             Only show the Build calls (do not execute the builds)"
+  echo "-v | --verbose              Show Logs in Plain Text (uses \`--progress=plain\`"
+  echo "-h | --help                 Print this Help."
+  echo
+  echo "----------------------------------------------------------------------------------------"
+}
+
+get_tag_from_version ()
+{
+  local  input=$1
+  arr=( ${input//\./ } ) # Split by .
+  arr="${arr[@]:0:4}" # Keep 1st 4 elements
+  arr="${arr// /.}" # Put it back together
+  result="${arr/+/-}" # Replace + with -
+  echo $result
+}
+
+# Setting Defaults
+POSITIONAL_ARGS=()
+NO_CACHE=false
+ARMORY_VERSION="$(python setup.py --version)"
```

Comment:
This is giving me ARMORY_VERSION 0.14.1.dev162+g768f824.d20220302134636 on noether and I cannot yet determine where that is coming from. 768f824 is the HEAD commit of use_scm right now.